### PR TITLE
[ADD] website_sale_firstname

### PR DESCRIPTION
**Pull request by trojikman**
This module adds the compatibility of partner_firstname module with website_sale,
i.e. allows you to use fields first name and last name in the form.

### CODE REVIEW

**Comment by yelizariev** [comment]
Runbot cannot find ``partner_firstname`` module

**Comment by yelizariev** [comment]
```
************* Module website_sale_firstname
website_sale_firstname/__init__.py:1: [C8202(unnecessary-utf8-coding-comment), ] UTF-8 coding is not necessary
************* Module website_sale_firstname.__manifest__
website_sale_firstname/__manifest__.py:1: [C8202(unnecessary-utf8-coding-comment), ] UTF-8 coding is not necessary
website_sale_firstname/__manifest__.py:5: [C8101(manifest-required-author), ] One of the following authors must be present in manifest: 'Odoo Community Association (OCA)'
************* Module website_sale_firstname.controllers.__init__
website_sale_firstname/controllers/__init__.py:1: [C8202(unnecessary-utf8-coding-comment), ] UTF-8 coding is not necessary
************* Module website_sale_firstname.controllers.main
website_sale_firstname/controllers/main.py:1: [C8202(unnecessary-utf8-coding-comment), ] UTF-8 coding is not necessary
```

**Comment by yelizariev** [comment]
There is still lint 

website_sale_firstname/__manifest__.py:6:80: E501 line too long (80 > 79 characters)

**Comment by yelizariev** [comment]
@etobella, could you please review this PR as the author of partner_firstname?

**Comment by etobella** [comment]
LGTM :+1: 
Just a question, it should be dependent on website_form_builder?

**Comment by yelizariev** [comment]
You probably mean website_form? We discussed it here: https://github.com/OCA/website/pull/514

**Comment by rafaelbn** [comment]
Hello!

@yelizariev in https://github.com/OCA/e-commerce/pull/245#issuecomment-423497093 @etobella means this amazing great module :smile: https://github.com/OCA/website/tree/11.0/website_form_builder

But the answer to @etobella is not, shouln't and it's nor needed. As the form builder take any field from the model you are using so if @etobella wants to have firstname in https://github.com/OCA/website/tree/11.0/website_form_builder then he need just to install https://github.com/OCA/crm/tree/11.0/crm_lead_firstname

I've read https://github.com/OCA/website/pull/514 and IMHO is:

- As website_sale depends in website_form we should add first , firstname in the form and not depending on website_sale. And after that add it in the form of website_sale
- I don't want to block anything and I like a lot your contributions but if we get this module (1) Has it any sense to add firstname in the checkout's form and not in the contact's form? (2) If tomorrow another contributor want to add firstname in the form without using website_sale we need another module and could be not compatible with this one

- website_form_firstname (should depend in partner_firstname + website_form)
- website_sale_firstname (just depend in website_form_firstname)

This is a really important module in Odoo website and Ecommerce in OCA so we should take care on it :octocat: 

@Yajo @chienandalu @eLBati what do you think about this? Thanks :yellow_heart:

**Comment by yajo** [comment]
I agree with @rafaelbn, `website_form_builder` fills the gap here. No need for an extra addon.

**Review by chienandalu — COMMENTED** [review]
I must agree as well

**Comment by yajo** [comment]
Closing then, feel free to reopen if needed. Thanks for the contribution anyway! :blush:

**Comment by yelizariev** [comment]
@Yajo are you sure that ``website_form_builder`` fills the gap? To make the field work on e-commerce form you have to redefine [_get_mandatory_billing_fields](https://github.com/OCA/website/pull/514/files#diff-fc9333806e42ae34a112c5a868995647R8). I don't see such code in ``website_form_builder``

**Comment by yajo** [comment]
I didn't see that part. Reopening then, thanks!

File: website_sale_firstname/README.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
+.. image:: https://img.shields.io/badge/license-LGPL--3-blue.png
```
Comment (simahawk) [inline website_sale_firstname/README.rst]: this is a new module, pls use new readme structure https://github.com/OCA/maintainer-tools/tree/master/template/module/readme

File: website_sale_firstname/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    "summary": """
+            Allow to change name field in form to first name and last name
+    """,
+    "version": "11.0.1.0.0",
+    "author": "IT-Projects LLC, "
+              "Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "maintainer": "IT-Projects LLC",
+    "category": "Extra Tools",
+    "website": "https://odoo-community.org/",
```
Comment (simahawk) [inline website_sale_firstname/__manifest__.py]: https://github.com/OCA/e-commerce

File: website_sale_firstname/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    "depends": [
+        "website_sale",
+        "partner_firstname",
+    ],
+    "data": [
+        "data/data.xml",
+    ],
+    "auto_install": False,
```
Comment (simahawk) [inline website_sale_firstname/__manifest__.py]: you can skip these flags

File: website_sale_firstname/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2018 Denis Mudarisov (IT-Projects LLC)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+
+
+class WebsiteSaleExtended(WebsiteSale):
+    def _get_mandatory_billing_fields(self):
+        fields = super()._get_mandatory_billing_fields()
+        fields = list(filter(lambda x: x != "name", fields))
+        return fields
```
Comment (simahawk) [inline website_sale_firstname/controllers/main.py]: I'd make this more readable like:
```python
return [
    fname for fname in super()._get_mandatory_billing_fields()
    if fname != 'name'
]
```

File: website_sale_firstname/controllers/main.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+    def _get_mandatory_shipping_fields(self):
+        fields = super()._get_mandatory_shipping_fields()
```
Comment (simahawk) [inline website_sale_firstname/controllers/main.py]: same :)

File: website_sale_firstname/data/data.xml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2018 Denis Mudarisov (IT-Projects LLC)
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <data>
```
Comment (simahawk) [inline website_sale_firstname/data/data.xml]: you can remove `data`

**Review by simahawk — CHANGES_REQUESTED** [review]
LG, thanks for your contrib! Minor things to change...

**Comment by yelizariev** [comment]
Requested changes are done and CI are green. We need one more approve, I guess

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**